File: Sum Root to Leaf Numbers.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
var summ int
func exploreNode(node *TreeNode, path string) {
    if node == nil {
        return
    }

    path += strconv.Itoa(node.Val)

    if node.Left == nil && node.Right == nil {
        i, _ := strconv.Atoi(path)
        summ += i
        return
    }

    exploreNode(node.Left, path)
    exploreNode(node.Right, path)
}

func sumNumbers(root *TreeNode) int {
    summ = 0
    exploreNode(root, "")
    return summ
}

File: Sum Root to Leaf Numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        int summ = 0;

        function<void(TreeNode*, string)> exploreNode;
        exploreNode = [&summ, &exploreNode] (TreeNode* node, string path) {
            if (node == nullptr) return;
            path += to_string(node->val);
            if (node->left == nullptr && node->right == nullptr) {
                summ += stoi(path);
                return;
            }
            exploreNode(node->left, path);
            exploreNode(node->right, path);
        };

        exploreNode(root, "");
        return summ;
    }
};
